examples/nextjs-with-typescript/pages/MediaChromeReactSimple.tsx

```diff
@@ -0,0 +1,343 @@
+          : MediaUIEvents.MEDIA_ENTER_PIP_REQUEST;
+        dispatch({ type });
+      }}
+    >
+      {!mediaIsPip ? 'Enter Pip' : 'Exit Pip'} React Only!
+    </button>
+  );
+};
+
+const FullscreenButton = () => {
+  const dispatch = useDispatch();
+  // @ts-ignore
+  const mediaIsFullscreen = useSelector((state) => state.mediaIsFullscreen);
+  return (
+    <button
+      style={{ cursor: 'pointer' }}
+      onClick={() => {
+        const type = mediaIsFullscreen
+          ? MediaUIEvents.MEDIA_EXIT_FULLSCREEN_REQUEST
+          : MediaUIEvents.MEDIA_ENTER_FULLSCREEN_REQUEST;
+        dispatch({ type });
+      }}
+    >
+      {!mediaIsFullscreen ? 'Enter Fullscreen' : 'Exit Fullscreen'} React Only!
+    </button>
+  );
+};
+
+const TimeRange = () => {
+  const dispatch = useDispatch();
+  // @ts-ignore
+  const mediaCurrentTime = useSelector((state) => state.mediaCurrentTime);
+  // @ts-ignore
+  const mediaDuration = useSelector((state) => state.mediaDuration);
+  return (
+    <div>
+      <span>Time Range</span>
+      <input
+        type="range"
+        min={0}
+        max={Number.isNaN(mediaDuration) ? 0 : mediaDuration}
+        value={mediaCurrentTime}
+        step={0.1}
+        onChange={(event) => {
+          const type = MediaUIEvents.MEDIA_SEEK_REQUEST;
+          const detail = +event.target.value;
+          dispatch({ type, detail });
+        }}
+      />
+    </div>
+  );
+};
+
+const VolumeRange = () => {
+  const dispatch = useDispatch();
+  // @ts-ignore
+  const mediaVolume = useSelector((state) => state.mediaVolume);
+  return (
+    <div>
+      <span>Volume Range</span>
+      <input
+        type="range"
+        min={0}
+        max={1}
+        value={mediaVolume}
+        step={0.1}
+        onChange={(event) => {
+          const type = MediaUIEvents.MEDIA_VOLUME_REQUEST;
+          const detail = +event.target.value;
+          dispatch({ type, detail });
+        }}
+      />
+    </div>
+  );
+};
+
+const Video = ({ src }: { src?: string }) => {
+  const mediaRefCallback = useMediaRefCallback();
+  return (
+    <video
+      ref={mediaRefCallback}
+      slot="media"
+      src={src}
+      preload="auto"
+      muted
+      crossOrigin=""
+      playsInline
+      // controls
+    >
+      <track
+        label="thumbnails"
+        default
+        kind="metadata"
+        src="https://image.mux.com/DS00Spx1CV902MCtPj5WknGlR102V5HFkDe/storyboard.vtt"
+      />
+      <track
+        label="English"
+        kind="captions"
+        srcLang="en"
+        src="./vtt/en-cc.vtt"
+      />
+    </video>
+  );
+};
+
+const MediaPlayButton = () => {
+  const dispatch = useDispatch();
+  // @ts-ignore
+  const mediaPaused = useSelector((state) => state.mediaPaused);
+  return (
+    // @ts-ignore
+    <media-play-button
+      mediapaused={mediaPaused ? '' : undefined}
+      // NOTE: The real version would actually forward on the mediaplayrequest/mediapauserequest events
```

Comment:
If you have to do all this wiring still, it doesn't seem like there's much value in using the web components like this. Right? Just the default styles maybe?

Reply:
In this example usage, yeah, it's definitely all for the styling (including e.g. swapping out the SVGs based on the `mediapaused` attr). This could also be the/a react wrapper we provide from a lib. In our first iteration, it sounds like we should treat these considerations as "out of scope" though.